src/modules/Builder/components/ConditionRow/Condition/SwitchCondition/SwitchCondition.tsx

```diff
@@ -65,8 +66,34 @@ export const SwitchCondition = ({
       payloadName,
       itemsName,
       itemName,
+      dateValueName,
     ],
   });
+
+  const parsedMinDateValue = minDateValue && new Date(minDateValue);
+  const parsedMaxDateValue = maxDateValue && new Date(maxDateValue);
+  const parsedDateValue = dateValue && new Date(dateValue);
+
+  useEffect(() => {
+    if (dateValue && typeof dateValue === 'string') {
+      setValue(dateValueName, new Date(dateValue));
+    }
+    if (minDateValue && typeof minDateValue === 'string') {
+      setValue(minDateValueName, new Date(minDateValue));
+    }
+    if (maxDateValue && typeof maxDateValue === 'string') {
+      setValue(maxDateValueName, new Date(maxDateValue));
+    }
```

Comment:
Could you add timezone when parsing the date, to prevent date shifting after selection?
example:
```
new Date(`${dateValue}T00:00:00`)
```

Reply:
done